lib/openid_connect/access_token.rb

```diff
@@ -21,7 +21,11 @@ def resource_request
       res = yield
       case res.status
       when 200
-        JSON.parse(res.body).with_indifferent_access
+        json = if res.headers["Content-Type"] =~ /application\/jwt/
+          JWT.decode(res.body, nil, false).first.with_indifferent_access
```

Comment:
I wonder if we could add:
```ruby
attr_optional algorithms, :jwks
```

```suggestion
          options = {}.tap do |hsh|
            hsh[:jkws] = { keys: jwks } if jwks
            hsh[:algorithms] = algorithms if algorithms
          end
          
          JWT.decode(res.body, nil, jwks.present?, options).first.with_indifferent_access
```